### ISSUE TITLE
`NO_COLOR` deprecation warning fires every time

### ISSUE THREAD

**Issue by gforsyth** [xonsh/xonsh#3834]
Even after replacing `NO_COLOR` with `RESET`, the warning still goes.
The warning also fires with a `xonsh --no-rc`, which seems incorrect.

## xonfig

<details>

```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.9.22.dev165        |
| Git SHA          | 580b675f             |
| Commit Date      | Sep 30 19:48:13 2020 |
| Python           | 3.8.5                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.7                |
| shell type       | prompt_toolkit       |
| pygments         | 2.6.1                |
| on posix         | True                 |
| on linux         | False                |
| on darwin        | True                 |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
| on jupyter       | False                |
| jupyter kernel   | None                 |
+------------------+----------------------+
```

</details>

## Expected Behavior
I should only get deprecation notices if I'm using a deprecated feature

## Current Behavior
The warning fires every time

## Steps to Reproduce

```
xonsh --no-rc
```

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

**Comment by scopatz** [comment]
Thanks for reporting this @gforsyth! I am seeing this too

**Comment by scopatz** [comment]
Potenial fix in  #3835

**Comment by gforsyth** [comment]
Closed via #3835
